File: 45-jump-game-ii/45-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
      
    

        jumpend = furthest = jump = 0 

        for i in range(len(nums)-1): 

            furthest = max(furthest , i+nums[i] ) 
            if i ==jumpend:
                jumpend = furthest 
                jump +=1 

            i+=1 

        return jump 

File: 45-jump-game-ii/45-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {

     
    int jumpend =0 ;
    int furthest =0; 
    int jump = 0 ; 
    int i = 0 ; 
    while(i < nums.size()-1 ) {
        
        furthest = max(furthest , i+nums[i] ) ; 
        if ( i ==jumpend){
            jumpend = furthest ;
            jump ++;  }

        i++; }

    return jump; 

        
        
        
    }
};